pkg/ccl/sqlproxyccl/balancer/balancer.go

```diff
@@ -162,6 +205,15 @@ func NewBalancer(
 	if options.timeSource == nil {
 		options.timeSource = timeutil.DefaultTimeSource{}
 	}
+	if options.rebalanceRate == 0 {
```

Comment:
nit: Instead of checking the options for default values, assign them in the struct and allow the options to override them. That makes for cleaner code and a more intuitive interface.

E.g. setting the default in the struct allows RebalanceDelay to accept 0 as the value that disables the delay feature instead of requiring a sentinel -1. 

```
func NewBalancer(...) {
    options := &balancerOptions {
        macConcurrentRebalances: defaultMaxConcurrentRebalances,
        timeSource: &timeutil.DefaultTimeSource{},
        rebalanceRate: defaultRebalanceRate
    } 
	for _, opt := range opts {
		opt(options)
	}
    // No if conditions necessary
}
```

I would also fold the default values into the struct initialization instead of defining constants.